src/Microsoft.AspNet.OData.Shared/Query/Expressions/AggregationBinder.cs

```diff
@@ -30,7 +30,7 @@ internal class AggregationBinder : ExpressionBinderBase
 
         private Type _groupByClrType;
 
-        private bool _linqToObjectMode = false;
+        private bool _classicEF = false;
```

Comment:
Note that we effectively switch the default here, since _linqToObjectMode has the same default value, but the opposite behavior, of _classicEF. That's fine, as long as bind is called to explicitly set this value. I assume there are no scenarios where this value is used without first calling bind?

Reply:
New default behavior is correct form .NET point of view. Expression like: object a = 15; must look like   object a = (object) 15; on MSIL to do proper boxing or AccessViolationException will be thrown. Unfortunately, EF6 can't translate that to SQL. So we have to do hack for EF6. The same for IEnumerable and IGroupingBy